Removed dev_timeout from junos_cli

codekuu opened the pull request:
Removed dev_timeout from junos_cli which passes None as a variable. max() does not like None variable which makes this input mandatory when using junos_cli in salt.

Did also merge the __timeout_decorator and __timeout_decorator_cleanargs into one with comments and more readable code.
This sovles the issue with problems running junos_cli and other commands where the __pubs are exposed and max() is breaking the function since dev_timeout is provided all the time (as None) even if not provided in cli.


### What does this PR do?
Fix issues related to napalm junos_cli and other methods.

### What issues does this PR fix or reference?
Fixes: #63599 #58108

### Previous Behavior
junos_cli was breaking due to timeout_decorator was provided with value None from the dev_timeout variable used in max()


### New Behavior
Working

### Merge requirements satisfied?
- [N/A] Docs
- [N/A] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [N/A] Tests written/updated

### Commits signed with GPG?
No

codekuu committed:
Removed dev_timeout from junos_cli 

Removed dev_timeout from junos_cli which passes None as a variable. max() does not like None variable which makes this input mandatory when using junos_cli in salt.

Fixes and related to #63599 #58108

codekuu committed:
Update junos.py

Fix issues related to timeout decorator breaking when dev_timeout or timeout variable is a none int.

Merged the timeout_decorator and timeout_decorator_cleankwargs

codekuu commented:
Tests and documentation is already correct for these fixes therefor no new tests or docs.

codekuu committed:
Update junos.py

codekuu committed:
Update junos.py

codekuu commented:
Pinging this

s0undt3ch commented on salt/modules/junos.py:
```diff
@@ -150,75 +150,50 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
+        # Run the function
+        try:
+            result = function(*args, **kwargs)
+        except:
+            raise
```
```suggestion
```
Don't catch exceptions just to raise them afterwards without doing anything with that information.

codekuu replied:
It was done like that before therefor I keept the same functionality.

s0undt3ch replied:
I have to disagree.
Neither [this](https://github.com/saltstack/salt/blob/3006.x/salt/modules/junos.py#L153-L169) chunk, nor [this](https://github.com/saltstack/salt/blob/3006.x/salt/modules/junos.py#L174-L203) has an exception caught just to be raised immediately.
There's code that runs, before raising the exception again.

Please remove that bare except, you're not doing anything with it.

codekuu committed:
Removed except raise

Removed except: raise block since it doesn't do anything. Keeping the finalyl since it adds back the kwargs that is needed and sets the timeout to the old timeout.

codekuu commented:
Removed the except: raise part but kept the finally

s0undt3ch commented on salt/modules/junos.py:
```diff
@@ -150,75 +150,48 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
     @wraps(function)
     def wrapper(*args, **kwargs):
+        timeout_changed = False
+        old_kwargs = kwargs
```
You actually need to create a copy, most likely, even a `copy.deepcopy` of the dictionary, otherwise changing `kwargs` will also change `old_kwargs`

s0undt3ch reviewed (changes requested):
We also need test coverage, the bare minimum is unit testing the decorator code path.
